SIMS/SIMS/Inventory.cs

```diff
@@ -0,0 +1,122 @@
+namespace SIMS;
+
+public class Inventory
+{
+    // Products property
+    private List<Product> Products { get; set; } = [];
+
+    // AddProduct method
+    public bool AddProduct(Product? product)
+    {
+        //check if the product is null
+        if (product is null)
+        {
+            return false;
+        }
+
+        //check if the product already exists
+        if (Products.Any(p => p.Name == product.Name))
+        {
+            return false;
+        }
+
+        //add the product to the list
+        Products.Add(product);
+        return true;
+    } // End of AddProduct method
+
+    //ViewAllProducts method
+    public void ViewAllProducts()
+    {
+        //check if there are no products
+        if (Products.Count == 0)
+        {
+            Console.WriteLine("No products found");
+        }
+
+        //print all the products
+        foreach (var product in Products)
+        {
+            Console.WriteLine(product);
+        }
+    } // End of ViewAllProducts method
+
+    //UpdateProduct method
+    public bool UpdateProduct(string currentProductName, string? newProductName,
+        decimal? newProductPrice, int? newProductQuantity)
```

Comment:
do u think it's a good practice to pass all product's params to this function? what if we have 10s of properties, think about more elegant way to handle the updation.